cmd/root_test.go

```diff
@@ -388,12 +391,44 @@ func TestRender(t *testing.T) {
 
 			var buf bytes.Buffer
 			err := Render(tt.input, tt.limit, &buf)
-			if tt.wantErr {
-				assert.Error(t, err)
+
+			if tt.json {
+				if !areJSONStringsEqual(tt.want.(string), buf.String()) {
+					t.Errorf("JSON output does not match the expected JSON")
+				}
 			} else {
-				assert.NoError(t, err)
+				if tt.wantErr {
+					assert.Error(t, err)
+				} else {
+					assert.NoError(t, err)
+				}
+				assert.Equal(t, tt.want, buf.String())
 			}
-			assert.Equal(t, tt.want, buf.String())
 		})
 	}
 }
+
+func areJSONStringsEqual(jsonStr1, jsonStr2 string) bool {
+	var data1 interface{}
+	var data2 interface{}
+
+	if err := json.NewDecoder(strings.NewReader(jsonStr1)).Decode(&data1); err != nil {
+		return false
+	}
+
+	if err := json.NewDecoder(strings.NewReader(jsonStr2)).Decode(&data2); err != nil {
+		return false
+	}
+
+	standardizedJSON1, err := json.Marshal(data1)
+	if err != nil {
+		return false
+	}
+
+	standardizedJSON2, err := json.Marshal(data2)
+	if err != nil {
+		return false
+	}
+
+	return reflect.DeepEqual(standardizedJSON1, standardizedJSON2)
+}
```

Comment:
I'm not sure we need this, have a look here at how you can add the new test:

https://github.com/BeirutOrg/gh-stars/pull/3/files

Reply:
mm, it will work, but not the best approach, as modifying the result format will necessitate test updates, and it's hard to detect the differences directly, especially if the differences are small (like one more space in the expected result).

And regarding the function above, I just realized that there is no need to marshal again to JSON in order to see if equal or not, so the function could be simplified to look like:
```
func areJSONStringsEqual(jsonStr1, jsonStr2 string) bool {
	var data1 interface{}
	var data2 interface{}

	if err := json.Unmarshal([]byte(jsonStr1), &data1); err != nil {
		return false
	}

	if err := json.Unmarshal([]byte(jsonStr2), &data2); err != nil {
		return false
	}

	return reflect.DeepEqual(data1, data2)
}
```

mm, another simpler option is to use `require.JSONEq` from `github.com/stretchr/testify/require`, I think that you will like it, since you won't need to make significant code updates, as we did previously, just this code: 
```
...
if tt.json {
    require.JSONEq(t, tt.want.(string), buf.String())
    return
}
...
```

I'm ready to update the PR based-on your comments, thanks for your response on my PR.

Reply:
> it will work, but not the best approach, as modifying the result format will necessitate test updates

I agree, except how many times will we update the results? :) 

Between ☝️ and adding a 3rd party dependency that we have to maintain (patches, new releases, etc.), I'd go with a hardcoded test

Reply:
I'd rather we invest any effort in improving the search algorithm which is far more important than this

Reply:
mm, yes I see what you mean, and I agree with you, but regarding the 3rd party dependency, it's already included in your project, you are using `github.com/stretchr/testify`, and `github.com/stretchr/testify/require` is a sub package of it.